src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/share/service/IntersectionComputer.java

```diff
@@ -201,6 +206,19 @@ public static List<PathSegment> calculatePathIntersectionFromDest(Collection<Flo
         return Lists.reverse(getLongestIntersectionOfSegments(pathSegmentIterators));
     }
 
+    private OverlappingSegmentsStats computeIntersectionCounters(Set<Edge> edges, Set<SwitchId> switches) {
+        int edgesOverlap = Sets.intersection(edges, targetPathEdges).size();
+        int switchesOverlap = Sets.intersection(switches, targetPathSwitches).size();
+        return new OverlappingSegmentsStats(edgesOverlap,
+                switchesOverlap,
+                percent(edgesOverlap, targetPathEdges.size()),
+                percent(switchesOverlap, targetPathSwitches.size()));
+    }
+
+    private int percent(int n, int from) {
+        return (int) ((n * 100.0f) / from);
```

Comment:
n = 1, from = 1000, gives 0%. This 0% is really different from 0% when n = 0. 
Also, division by 0 is not handled here.

Reply:
It is the same situation, it is the old code, I will fix it. Thanks!

Reply:
Regarding our configuration max allowed depth is 35, so the situation when you 1/1000 is unreal in our current environment.